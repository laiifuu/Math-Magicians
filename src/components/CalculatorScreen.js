import React from 'react';
import PropTypes from 'prop-types';

class CalculatorScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { total, next, operation } = this.props;
    return (<div className="width_100 black left-text">{`${total || ''} ${operation || ''} ${next || ''}`}</div>);
  }
}

CalculatorScreen.propTypes = {
  total: PropTypes.string,
  next: PropTypes.string,
  operation: PropTypes.string,
};

CalculatorScreen.defaultProps = {
  total: '',
  next: '',
  operation: '',
};

export default CalculatorScreen;
