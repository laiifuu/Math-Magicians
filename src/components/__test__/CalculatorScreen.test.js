import renderer from 'react-test-renderer';
import CalculatorScreen from '../CalculatorScreen';

describe('Testing CalculatorScreen component', () => {
  it('renders CalculatorScreen component correctly', () => {
    const tree = renderer
      .create(<CalculatorScreen total="0" operation="" next="" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
