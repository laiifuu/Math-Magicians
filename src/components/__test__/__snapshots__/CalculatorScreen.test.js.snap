// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CalculatorScreen component renders CalculatorScreen component correctly 1`] = `
<div
  className="width_100 black left-text"
>
  0  
</div>
`;
